src/x.c

```diff
@@ -628,21 +628,19 @@ bool x_validate_pixmap(xcb_connection_t *c, xcb_pixmap_t pixmap) {
 	free(r);
 	return ret;
 }
-/// Names of root window properties that could point to a pixmap of
-/// background.
-static const char *background_props_str[] = {
-    "_XROOTPMAP_ID",
-    "_XSETROOT_ID",
-    0,
-};
+
+/// A list of root window properties sorted by the probability of pointing to the
+/// background pixmap. See
+/// https://metacpan.org/pod/X11::Protocol::XSetRoot#ROOT-WINDOW-PROPERTIES for details.
+static const char *background_props_str[] = {"_XROOTPMAP_ID", "ESETROOT_PMAP_ID"};
```

Comment:
is `_XSETROOT_ID` not thing anymore?

Reply:
i think it's definitely still being used, we should keep it. i also don't think documentation for a perl module is a good reference...

Reply:
> is `_XSETROOT_ID` not thing anymore?

as far as i know and understand, it never was a thing in terms of pointing to the background pixmap. it's used to store colors associated with the background pixmap and to indicate presence of the background pixmap.

there are two known troublemakers that use this property:
1) xsetroot seems to use it only if the root window's visual class is dynamic (i have no idea who may have it so i was unable to make it set the `_XSETROOT_ID` property no matter how hard i tried) the same way as described above;
2) display seems to always use it the same way as described above and that's what we're getting when trying to use the pixmap `_XSETROOT_ID` points to:
```
...
[ 06/17/2023 23:18:55.622 root_damaged FATAL ERROR ] root back pixmap geometry:
[ 06/17/2023 23:18:55.622 root_damaged FATAL ERROR ]   x, y: 0, 0
[ 06/17/2023 23:18:55.622 root_damaged FATAL ERROR ]   width, height: 1, 1
[ 06/17/2023 23:18:55.622 root_damaged FATAL ERROR ]   border_width: 0
[ 06/17/2023 23:18:55.622 root_damaged FATAL ERROR ]   depth: 1
[ 06/17/2023 23:18:55.622 root_damaged FATAL ERROR ]   root: 0x0001e0
[ 06/17/2023 23:18:55.622 root_damaged FATAL ERROR ] trying to bind it...
[ 06/17/2023 23:18:55.622 x_create_picture_with_pictfmt_and_pixmap ERROR ] failed to create picture (X error 8 MATCH request 139 minor 4 serial 2072)
[ 06/17/2023 23:18:55.622 root_damaged ERROR ] Failed to bind root back pixmap
...
```

see for details:
https://gitlab.freedesktop.org/xorg/app/xsetroot/-/blob/master/xsetroot.c#L530-533
https://github.com/ImageMagick/ImageMagick/blob/main/MagickCore/xwindow.c#L9203-L9228
https://github.com/ImageMagick/ImageMagick/blob/main/MagickCore/xwindow.c#L1853-L1878

i have never met an `_XSETROOT_ID` property solely pointing to the background pixmap and in my opinion it's better to ignore it because an attempt to use it most likely will result in an error shown above.

> i think it's definitely still being used

it's indeed still being used, but it's used not in a way we expect it to be used and never was.

> i also don't think documentation for a perl module is a good reference...

since none of these properties are standard it's the best thing that we have :shrug:

however, if you insist, i can add the `_XSETROOT_ID` property in the end of the list to be used as a last resort.

Reply:
ok, let me read up on the `_XSETROOT_ID`. the lack of standardization is indeed annoying.

Reply:
ok, cool, you are right. as for referencing, I suggest have a brief explanation, and link to some of the application to provide proof that's how it's used:

```c
/// A list of root window properties sorted by the probability of pointing to the
/// background pixmap. 
///
/// Note `_XSETROOT_ID` is also a pixmap created by wallpaper utilities, however
/// it's not the actual root background pixmap. It's instead a dummy pixmap to
/// keep the colourmap entry for the background pixmap alive.
///
/// See:
///   https://github.com/ImageMagick/ImageMagick/blob/main/MagickCore/xwindow.c?rgh-link-date=2023-06-17T20%3A53%3A47Z#L1853-L1878
///   https://metacpan.org/pod/X11::Protocol::XSetRoot#ROOT-WINDOW-PROPERTIES
///   https://gitlab.freedesktop.org/xorg/app/xsetroot/-/blob/master/xsetroot.c
/// for examples of how these properties are used.
```